### PR TITLE
profile list instructions

### LINKED ISSUE #1085: profileList only mentioned in the configuration reference (schema.yaml)
We should add some narrative surrounding the use of profileList in the documentation, and perhaps not only have it in the reference section without mentioning it anywhere else.

(Idea by @yuvipanda, and I'm writing it down as an issue to not loose track of it)

### PR DESCRIPTION
**Pull request by choldgraf**
This adds a short section on configuring the profile list for users. It currently
uses the passthrough configuration, though I think I ran into a bug (#1097).

closes #1085 closes #1083

### CODE REVIEW

File: doc/source/user-environment.rst [hunk 1]
```diff
@@ -293,6 +296,103 @@ across sessions. To resolve this, take the following steps:
...
+Docker images, to select the hardware on which they want their jobs to run,
+or to configure default interfaces such as Jupyter Lab vs. RStudio.
+
+Each configuration is a set of options for `Kubespawner <https://github.com/jupyterhub/kubespawner>`_,
+which defines how Kubernetes should launch a new user server pod. Any
+configuration options passed to the `profileList` configuration will
+overwrite the defaults in Kubespawner (or any configuration you've
+added elsewhere in your helm chart).
+
+Profiles are stored under ``singluser: profileList:``, and are defined as
```
Comment (consideRatio) [inline doc/source/user-environment.rst]:
```suggestion
Profiles are stored under ``singluser.profileList``, and are defined as
```

File: doc/source/user-environment.rst [hunk 2]
```diff
@@ -293,6 +296,103 @@ across sessions. To resolve this, take the following steps:
...
+         kubespawner_override:
+           image: jupyter/datascience-notebook:2343e33dec46
+       - display_name: "Spark environment"
+         description: "The Jupyter Stacks spark image!"
+         kubespawner_override:
+           image: jupyter/all-spark-notebook:2343e33dec46
+
+This allows users to select from three profiles, each with their own
+environment (defined by each Docker image in the configuration above). This is
+equivalent to the following configuration using `extraCongig` to directly define
```
Comment (consideRatio) [inline doc/source/user-environment.rst]:
```suggestion
equivalent to the following configuration using `hub.extraConfig` to directly define
```

File: doc/source/user-environment.rst [hunk 3]
```diff
@@ -293,6 +296,103 @@ across sessions. To resolve this, take the following steps:
...
+configuration for KubeSpawner:
+
+.. code-block:: yaml
+
+   singleuser:
+     image:
+       name: jupyter/minimal-notebook
+       tag: 2343e33dec46
+   hub:
+     extraConfig: |
```
Comment (consideRatio) [inline doc/source/user-environment.rst]: While it is still supported to write a string directly to `hub.extraConfig`,  tend to do it like this instead now days. I avoid some warnings about conversion etc. that bugged me as the default value of `hub.extraConfig` is a `{}` and overwriting a dict with a string results in a warning if I recall correctly.

```yaml
hub:
  extraConfig:
    myProfiles: |
      ...
```

@minrk / @yuvipanda got input on this? Examplify with dict or string?

Comment (choldgraf) [reply doc/source/user-environment.rst]: I'm happy with whatever you all think is the best practice to recommend to people (we could also remove this section if we're confident that the passthrough works now)

**Comment by choldgraf** [comment]
I just gave another shot at the passThrough config for profileList using the latest-published helm chart, and it works! So I think we can now merge this without feeling like we're misleading people :-)

re: @consideRatio 's comments about the `extraConfig` stuff above - I'm also fine just removing that addendum, since the pass-through seems to be working now!

**Comment by consideRatio** [comment]
@choldgraf thanks for the work on this!!

**Comment by choldgraf** [comment]
oops - accidental close - latest push removes that "extraConfig" section at the end. This is good to go from my end

**Pull request merged**